lib/logsource.c

```diff
@@ -242,6 +247,9 @@ log_source_post(LogSource *self, LogMessage *msg)
 
   old_window_size = g_atomic_counter_exchange_and_add(&self->window_size, -1);
 
+  if (G_UNLIKELY(g_atomic_counter_get(&self->window_size) == 0))
+    msg_debug("Source has been suspended", evt_tag_str("group_name", self->options->group_name));
```

Comment:
Oh I see, I'll use the `old_window_size` in this case.

AFAIK `log_source_flow_control_suspend()` is only called by the the ack tracker when `LogQueue` drops a message with `AT_SUSPENDED` (when the queue is full).

Usually, the `window_size` reaches 0 before the queue is completely filled, this is why I placed this
debug message here.